src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -296,35 +308,41 @@ module.exports = function<T, P, I, TI, C>(
       current.alternate.child = null;
       current.alternate.return = null;
     }
+    return noErrors;
   }
 
-  function commitUnmount(current : Fiber) : void {
+  // Returns true if it completed without any errors
+  function commitUnmount(current : Fiber) : boolean {
+    let noErrors = true;
     switch (current.tag) {
       case ClassComponent: {
         detachRef(current);
         const instance = current.stateNode;
         if (typeof instance.componentWillUnmount === 'function') {
           const error = tryCallComponentWillUnmount(instance);
           if (error) {
-            captureError(current, error, true);
+            captureError(current, error);
+            noErrors = false;
           }
         }
-        return;
+        break;
       }
       case HostComponent: {
         detachRef(current);
```

Comment:
Good call, I think those should be special-cased like cWU so they don't interrupt deletion.